worker/src/lib.rs

```diff
@@ -0,0 +1,504 @@
+/// the `ResponseData` associated type to know the datatype to send to the `worker_response`
+/// method.
+pub struct ResponseHandler<P: Worker> {
+    /// function provided by the host to send response to `run()`
+    response_function: lv2_sys::LV2_Worker_Respond_Function,
+    /// Response handler provided by the host, must be passed to the host provided
+    /// response_function.
+    respond_handle: lv2_sys::LV2_Worker_Respond_Handle,
+    phantom: PhantomData<P>,
+}
+
+impl<P: Worker> ResponseHandler<P> {
+    /// Allow to give response to the `run` context.
+    ///
+    /// This method allow to give a response to the `run` context. After calling this method, the host will call
+    /// `worker_response` and give the passed data.
+    ///
+    /// **Notes about the passed data:** The buffer used to pass data is managed by the host. That
+    /// mean the size is unknown and may be limited. So if you need to pass huge amount of data,
+    /// it's preferable to use another way, for example a sync::mpsc channel.
+    pub fn respond(&self, response_data: P::ResponseData) -> Result<(), WorkerError>
+    where
+        P::WorkData: 'static + Send,
+    {
+        unsafe {
+            let response_data = mem::ManuallyDrop::new(response_data);
+            let size = mem::size_of_val(&response_data) as u32;
+            let ptr = &response_data as *const _ as *const c_void;
+            let response_function = if let Some(response_function) = self.response_function {
+                response_function
+            } else {
+                return Err(WorkerError::Unknown);
+            };
+            match (response_function)(self.respond_handle, size, ptr) {
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_SUCCESS => Ok(()),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_UNKNOWN => Err(WorkerError::Unknown),
+                lv2_sys::LV2_Worker_Status_LV2_WORKER_ERR_NO_SPACE => Err(WorkerError::NoSpace),
+                _ => Err(WorkerError::Unknown),
+            }
+        }
+    }
+}
+
+/// Errors potentially generated by [`Worker`](trait.Worker.html) methods
+pub enum WorkerError {
+    /// Unknown or general error
+    Unknown,
+    /// Failure due to a lack of space
+    NoSpace,
+}
+
+/// The trait providing worker extension to rust-lv2 plugins.
+///
+/// This trait, and `struct` that implements it, provide methods called by the host to implement
+/// the worker extension.
+///
+/// However, the host will not directly use these methods. Instead, the host use wrapping functions
+/// associated with [`WorkerDescriptor`](struct.WorkerDescriptor.html) and then, you pass this
+/// functions to the host with `match_extensions!` macro in `extension_data()` methods of the
+/// Plugin trait.
+pub trait Worker: Plugin {
+    /// Data sended to the worker thread
+    type WorkData: 'static + Send;
+    /// Data sended by the worker thread to `work_response`
+    type ResponseData: 'static + Send;
```

Comment:
These types have to be `Copy` too, since the host will copy their data byte by byte before sending it to the worker thread.
```suggestion
    /// Data sent to the worker thread
    type WorkData: 'static + Copy + Send;
    /// Data sent by the worker thread to `work_response`
    type ResponseData: 'static + Copy + Send;
```